go/types/sequence_cursor.go

```diff
@@ -4,24 +4,41 @@
 
 package types
 
-import "github.com/attic-labs/noms/go/d"
+import (
+	"math"
+	"os"
+	"runtime"
+
+	"github.com/attic-labs/noms/go/d"
+)
 
 // sequenceCursor explores a tree of sequence items.
 type sequenceCursor struct {
-	parent *sequenceCursor
-	seq    sequence
-	idx    int
+	parent        *sequenceCursor
+	seq           sequence
+	idx           int
+	readAheadMap  map[int]chan sequence
+	readAheadNext int
 }
 
+// newSequenceCursor creates a cursor on seq positioned at idx.
+// If idx < 0, count backward from the end of seq.
 func newSequenceCursor(parent *sequenceCursor, seq sequence, idx int) *sequenceCursor {
 	d.PanicIfTrue(seq == nil)
 	if idx < 0 {
 		idx += seq.seqLen()
 		d.PanicIfFalse(idx >= 0)
 	}
 
-	cur := &sequenceCursor{parent, seq, idx}
-	return cur
+	var readAheadMap map[int]chan sequence
+	// enable read-ahead if the cursor represents to a meta-sequence containing leaves
+	ms, ok := seq.(metaSequence)
+	if os.Getenv("NOMS_NO_READAHEAD") == "" && ok {
```

Comment:
I remember us talking about this back in the original code review, but an environment variable here isn't great. Especially since we should port this code to JS, and we can't access environment variables from JS without some contortions (webpack tricks).

When do you expect this to be used? If there is actually a use case immediately, can we make it a global variable or something and set it from clients. Like, with a command line argument even.

If it's not needed immediately, can we just take it out, and if it's needed later, add something like it back?